docs/source/configuration/instruments.mdx

```diff
@@ -0,0 +1,310 @@
+description: Create and customize instruments to collect data and report measurements from the Apollo Router's pipeline services.
+---
+
+An **instrument** is used to collect data and report measurements to a metric backend. The Apollo Router supports the following metric instruments: 
+
+* Counter
+* Gauge
+* Histogram 
+
+Instruments in the router are configurable in `router.yaml` as `telemetry.instruments`.
+
+### Standard instruments
+
+OpenTelemetry semantic conventions define several standard metric instruments that are available and configurable in the router:
+
+* `http.server.active_requests` - The number of active requests in flight.
+* `http.server.request.body.size` - A histogram of request body sizes for requests handled by the router.
+* `http.server.request.duration` - A histogram of request durations for requests handled by the router.
+* `http.server.response.body.size` - A histogram of response body sizes for requests handled by the router.
+
+For more information, see the [Open Telemetry semantic conventions for HTTP metrics](https://opentelemetry.io/docs/specs/semconv/http/http-metrics/).
+
+These standard instruments are configurable in `router.yaml`:
+
+```yaml title="router.yaml"
+telemetry:
+  instruments:
+    router:
+      http.server.active_requests: true # (default false)
+      http.server.request.body.size: true # (default false)
+      http.server.request.duration: true # (default false)
+      http.server.response.body.size: true # (default false)
+```
+
+Standard instruments can be customized by attaching or removing attributes.
+
+```yaml title="router.yaml"
+telemetry:
+  instruments:
+    default_attribute_requirement_level: required
+    router:
+      http.server.active_requests: 
+        attributes:
+          # Standard attributes
+          http.response.status_code: false
+          # Custom attribute
+          "acme.my_attribute":
+            response_header: "x-my-header"
+```
+
+See the [attributes](#attributes) configuration for more information.
+
+### Custom instruments
+
+You can define custom instruments on the router, supergraph and subgraph services in the router pipeline.
+
+When defining a custom instrument, make sure to reference the [OpenTelemetry semantic conventions](https://opentelemetry.io/docs/specs/semconv/general/metrics/), as it will help guide you to:
+
+
+
+In the example configuration below, three custom instruments are defined (`acme.request.duration`, `acme.graphql.requests`, `acme.graphql.subgraph.errors`), one for each service of the router pipeline (`router`, `supergraph`, `subgraph`):
+
+```yaml title="router.yaml"
+telemetry:
+  instruments:
+    router:
+      http.server.active_requests: true
+      acme.request.duration:
+        value: duration
+        type: counter
+        unit: kb
+        description: "my description"
+        attributes:
+          http.response.status_code: true
+          "my_attribute":
+            response_header: "x-my-header"
```

Comment:
Not really part of writing the docs, but how will this work? Are we documenting source modifications to add support for metrics somehow? (Feel free to ignore this comment or point me at design docs).

Reply:
Metrics can largely be driven by yaml at this point. If users need to do something custom then they can put something into context using rhai and then configure the metric to pick that up.